test/client/query-subscribe.js

```diff
@@ -11,586 +11,579 @@ module.exports = function(options) {
+    async.parallel([
+      function(cb) {
+        connection1.get('dogs', 'fido').on('error', done).create({age: 3}, cb);
+      },
+      function(cb) {
+        connection1.get('dogs', 'spot').on('error', done).create({age: 5}, cb);
+      }
+    ], function(err) {
+      if (err) return done(err);
+      var query = connection2.createSubscribeQuery('dogs', matchAllDbQuery, null, function(err) {
+        if (err) return done(err);
+        connection1.get('dogs', 'fido').del();
       });
-
-      it('sends agentCustom to the db\'s getSnapshot call', function(done) {
-        var query = connection.createSubscribeQuery('dogs', this.matchAllDbQuery);
-        var getSnapshotSpy = sinon.spy(this.backend.db, 'getSnapshot');
-
-        query.on('insert', function() {
-          // The first call to getSnapshot is when the document is created
-          // The seconds call is when the event is triggered, and is the one we are testing here
-          expect(getSnapshotSpy.callCount).to.equal(2);
-          expect(getSnapshotSpy.getCall(1).args[3]).to.deep.equal(expectedArg);
-          done();
-        });
-        connection.get('dogs', 'fido').create({age: 3});
+      query.on('error', done);
+      var removed = false;
+      connection2.get('dogs', 'fido').on('del', function() {
+        expect(removed).equal(true);
+        done();
       });
-
-      it('sends agentCustom to the db\'s getSnapshotBulk call', function(done) {
-        // Ensures that getSnapshotBulk is called, instead of getSnapshot
-        this.backend.db.canPollDoc = function() {
-          return false;
-        };
-
-        var query = connection.createSubscribeQuery('dogs', this.matchAllDbQuery);
-        var getSnapshotBulkSpy = sinon.spy(this.backend.db, 'getSnapshotBulk');
-
-        query.on('insert', function() {
-          expect(getSnapshotBulkSpy.callCount).to.equal(1);
-          expect(getSnapshotBulkSpy.getCall(0).args[3]).to.deep.equal(expectedArg);
-          done();
-        });
-        connection.get('dogs', 'fido').create({age: 3});
+      query.on('remove', function(docs, index) {
+        removed = true;
+        expect(util.pluck(docs, 'id')).eql(['fido']);
+        expect(util.pluck(docs, 'data')).eql([{age: 3}]);
+        expect(index).a('number');
+        var results = util.sortById(query.results);
+        expect(util.pluck(results, 'id')).eql(['spot']);
+        expect(util.pluck(results, 'data')).eql([{age: 5}]);
       });
     });
+  });
 
-    it('changing a filtered property removes from a subscribed query', function(done) {
-      var connection = this.backend.connect();
-      async.parallel([
-        function(cb) {
-          connection.get('dogs', 'fido').on('error', done).create({age: 3}, cb);
-        },
-        function(cb) {
-          connection.get('dogs', 'spot').on('error', done).create({age: 3}, cb);
-        }
-      ], function(err) {
+
+  it('creating an additional document updates a subscribed query', function(done) {
```

Comment:
We can get rid of this second one, apparently we've had this exact duplicate copy for a long time.

Reply:
Removed